tools/monitoring/criersub/subscriber_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package criersub
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+
+	"cloud.google.com/go/pubsub"
+)
+
+type contextKey int
+
+const (
+	keyError contextKey = iota
+)
+
+type fakeSubscriber struct {
+	SubscriberOperation
+	name string
+}
+
+func getFakeSubscriber(n string) *SubscriberClient {
+	return &SubscriberClient{&fakeSubscriber{name: n}}
+}
+
+func (fs *fakeSubscriber) Receive(ctx context.Context, f func(context.Context, *pubsub.Message)) error {
+	if err := ctx.Value(keyError); err != nil {
+		return err.(error)
+	}
+	return nil
+}
+
+func (fs *fakeSubscriber) String() string {
+	return fs.name
+}
+
+func TestSubscriberClient_ReceiveMessageAckAll(t *testing.T) {
+	receivedMsgs := make([]*ReportMessage, 3)
+
+	type arguments struct {
+		ctx context.Context
+		f   func(*ReportMessage)
+	}
+	tests := []struct {
+		name string
+		args arguments
+		want error
+	}{
+		{
+			name: "Message Received",
+			args: arguments{
+				ctx: context.Background(),
+				f: func(message *ReportMessage) {
+					receivedMsgs = append(receivedMsgs, message)
```

Comment:
append will add to the slice. So, it will increase the size of the slice. Do you want recdMsg[i]?

Reply:
Updated.